pkg/client/cache/watch_cache.go

```diff
@@ -27,12 +27,22 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+// WatchCacheEvent is a single "watch event" that is send to users of
+// WatchCache. Additionally to a typical "watch.Event" it contains
+// the previous value of the object to enable proper filtering in the
+// upper layers.
```

Comment:
OK - I remember now why it is here - there is a reference to WatchCache in reflector.go:
https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/reflector.go#L238

The fix for it is to change the signature of Replace() method in Store interface to take "resource version" and get rid of this hack.
I would like to avoid fixing it in this PR, so I'm adding a TODO and will fix that in a separate PR later this week.